nautobot_device_lifecycle_mgmt/jobs/cve_tracking.py

```diff
@@ -65,3 +70,219 @@ def run(self, data, commit):  # pylint: disable=too-many-locals
+            software_cve_info = self.get_cve_info(cpe_software_search_url, software.id)
+
+            cve_counter += len(software_cve_info)
+            self.create_dlc_cves(software.id, software_cve_info)
+
+        self.log_success(
+            message=f"""Performed discovery on all software meeting
+                        naming standards.  Added {cve_counter} CVE."""
+        )
+        self.update_cves()
+
+    def validate_platform_name(self, platform: str) -> str:
+        """Platform names are typically entered with a preceding manufacturername.
+
+        Remove the preceding Manufacturer name and return only a platform
+        by splitting on the first space and replace spaces with underscores to
+        prepare.
+        """
+        platform = platform.split(" ", 1)[1].lower()
+        platform = platform.replace(" ", "_")
+
+        return platform
+
+    def validate_platform_version(self, version: str) -> str:
+        """Platform versions may contain characters that need to be escaped.
+
+        Create a compatible version.
+        If any additional chars need to be escaped, add to the list.
+        """
+        escape_list = [r"\(", r"\)"]
+        for escape_char in escape_list:
+            if re.search(escape_char, version):
+                version = re.sub(escape_char, "\\" + escape_char, version)
+
+        return version
+
+    def create_cpe_software_search_url(self, manufacturer: str, platform: str, version: str) -> str:
+        """Return the url for a cpe search against the NIST DB."""
+        base_url = (
+            f"""https://services.nvd.nist.gov/rest/json/cpes/1.0?addOns=cves&cpeMatchString=cpe:2.3:*:"""  # noqa: F541
+        )
+        extended_url = f"{manufacturer}:{platform}:{version}:*:*:*:*:*:*:*"
+
+        return f"{base_url}{extended_url}"
+
+    def prep_cve_for_dlc(self, url: str) -> dict:
+        """Performs search of NIST CVE DB and prepares data for insertion to DLC Management Plugin."""
+        cve_name = url.split("/")[-1]
+        cve_search_url = f"{url}"
+        result = json.loads(requests.get(cve_search_url).text)
+
+        if result.get("message"):
+            self.log_info(message=f"""CVE {cve_name} DOES NOT EXIST IN NIST DATABASE""")
+            return
+
+        cve_base = result["result"]["CVE_Items"][0]
+        cve_description = cve_base["cve"]["description"]["description_data"][0]["value"]
+        cve_published_date = cve_base.get("publishedDate")
+        cve_modified_date = cve_base.get("lastModifiedDate")
+        cve_impact = cve_base.get("impact")
+
+        reference_data = result["result"]["CVE_Items"][0]["cve"]["references"]["reference_data"]
+        if len(reference_data) > 0:
+            cve_url = reference_data[0].get("url", f"https://www.cvedetails.com/cve/{cve_name}/")
+        else:
+            cve_url = f"https://www.cvedetails.com/cve/{cve_name}/"
```

Comment:
```suggestion
        cve_url = f"https://www.cvedetails.com/cve/{cve_name}/"
        if reference_data and reference_data[0].get('url'):
            cve_url = reference_data[0]["url"])
```

Reply:
In general, I suspect you can use this pattern:

`if my_list:` vs `if len(my_list) > 0` 

There may be edge cases I am not considering, but for the most part,  if `my_list = []`, that returns falsey on a conditional check.